common/e2e.go

```diff
@@ -70,5 +70,24 @@ func RunE2ETests(t *testing.T, cfg *config.Config) {
 	if !cfg.DryRun {
 		log.Println("Running e2e tests...")
 		ginkgo.RunSpecsWithDefaultAndCustomReporters(t, "OSD e2e suite", []ginkgo.Reporter{reporter})
+		// upgrade cluster if requested
```

Comment:
We have to shuffle around some logic (upgrade, destroy) to put it where we call the ginkgo tests. That way it's very clean/clear (not a dryRun? run tests. upgrade? upgrade. run tests. destroy cluster? destroy cluster.)